Misleading error message when creating a pin on non existent branch

When attempting to create a pin on a branch which does not exist, such as shown below, the resulting error message: “Git repository seems just initialized, try again after your first commit” isn't correct.
```sh
> opam pin -y https://github.com/MagnusS/okra.git#foo
[ERROR] Could not synchronize
        /tmp/opam-pin-cache-1171957-524fbc/515c93b7e721deaa from
        "git+https://github.com/MagnusS/okra.git#foo":
        Git repository seems just initialized, try again after your first
        commit
[ERROR] Could not retrieve git+https://github.com/MagnusS/okra.git#foo
```

Output of `opam config report`:
```sh
admin:main>opam config report
# opam config report
# opam-version         2.1.2 
# self-upgrade         no
# system               arch=x86_64 os=linux os-distribution=ubuntu os-version=22.04
# solver               builtin-mccs+glpk
# install-criteria     -removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed
# upgrade-criteria     -removed,-count[avoid-version,changed],-count[version-lag,solution],-count[missing-depexts,changed],-new
# jobs                 15
# repositories         1 (http), 2 (local) (default repo at 5c22df21)
# pinned               3 (git)
# current-switch       /home/cuihtlauac/tarides/admin
# ocaml:native         true
# ocaml:native-tools   true
# ocaml:native-dynlink true
# ocaml:stubsdir       /home/cuihtlauac/tarides/admin/_opam/lib/ocaml/stublibs:/home/cuihtlauac/tarides/admin/_opam/lib/ocaml
# ocaml:preinstalled   false
# ocaml:compiler       4.14.0
[WARNING] PATH contains '.' : this is a likely cause of trouble.
```

--- a/master_changes.md
+++ b/master_changes.md
@@ -221,6 +221,7 @@ users)
   * git: disable colored output [#4884 @rjbou]
   * Check if a source is up to date with subpath [#4876 @rjbou]
   * git, hg: Use the full SHA1 revision instead of just the 8 first characters [#5342 @reynir]
+  * Git: differentiate non initialised repo and branch not found errors [#5326 @rjbou - fix #5324]
 
 ## Build
   * Bump vendored base64 to 3.5.1 to fix compilation on OCaml >= 5.0 in vendored mode [#5464 @deech]

--- a/src/client/opamPinCommand.ml
+++ b/src/client/opamPinCommand.ml
@@ -398,7 +398,7 @@ let fetch_all_pins st ?working_dir pins =
   if errored = []
   || OpamConsole.confirm
        "Could not retrieve some package sources, they will not be pinned nor \
-        installed:%s\n\
+        installed:\n%s\n\
         Continue anyway?"
        (OpamStd.Format.itemize (fun p ->
             Printf.sprintf "%s:%s"

--- a/src/repository/opamGit.ml
+++ b/src/repository/opamGit.ml
@@ -139,14 +139,12 @@ module VCS : OpamVCS.VCS = struct
             else failwith "Commit not found on repository"))
       else
       let error = r in
-      git repo_root ["ls-files"] @@> function
+      git repo_root ["branch"; "-a"] @@> function
       | { OpamProcess.r_code = 0; OpamProcess.r_stdout = []; _ } ->
-        git repo_root ["show"] @@> fun r ->
-        if OpamProcess.is_failure r then
-          failwith "Git repository seems just initialized, \
-                    try again after your first commit"
-        else
-          OpamSystem.process_error error
+        failwith "Git repository seems just initialized, \
+                  try again after your first commit"
+      | { OpamProcess.r_code = 0; OpamProcess.r_stdout = _::_; _ } ->
+        failwith (Printf.sprintf "Branch %s not found" branch)
       | _ -> OpamSystem.process_error error
 
   let revision repo_root =

--- a/tests/reftests/pin.test
+++ b/tests/reftests/pin.test
@@ -257,7 +257,7 @@ pin-empty is now pinned to git+file://${BASEDIR}/pin-empty#master (version dev)
 Ok, pin-empty is no longer pinned to git+file://${BASEDIR}/pin-empty#master (version dev)
 ### opam pin --no-action ./pin-empty#inexistant
 [ERROR] Could not synchronize ${BASEDIR}/OPAM/pinning/.opam-switch/sources/pin-empty from "git+file://${BASEDIR}/pin-empty#inexistant":
-        Git repository seems just initialized, try again after your first commit
+        Branch inexistant not found
 Could not retrieve some package sources, they will not be pinned nor installed:
   - pin-empty:git+file://${BASEDIR}/pin-empty#inexistant
 

--- a/tests/reftests/pin.test
+++ b/tests/reftests/pin.test
@@ -234,3 +234,32 @@ The following actions will be performed:
 -> removed   qux.dev
 Done.
 ### opam pin
+### : Error retrieving git pin :
+### mkdir pin-empty
+### git -C pin-empty init -q --initial-branch=master
+### git -C pin-empty config core.autocrlf false
+### <pin:pin-empty/opam>
+opam-version: "2.0"
+### opam pin --no-action ./pin-empty
+[ERROR] Could not synchronize ${BASEDIR}/OPAM/pinning/.opam-switch/sources/pin-empty from "git+file://${BASEDIR}/pin-empty#master":
+        Git repository seems just initialized, try again after your first commit
+Could not retrieve some package sources, they will not be pinned nor installed:
+  - pin-empty:git+file://${BASEDIR}/pin-empty#master
+
+Continue anyway? [y/n] y
+### opam pin
+### git -C pin-empty add -A
+### git -C pin-empty commit -qm 'opam'
+### opam pin --no-action ./pin-empty
+Package pin-empty does not exist, create as a NEW package? [y/n] y
+pin-empty is now pinned to git+file://${BASEDIR}/pin-empty#master (version dev)
+### opam unpin pin-empty
+Ok, pin-empty is no longer pinned to git+file://${BASEDIR}/pin-empty#master (version dev)
+### opam pin --no-action ./pin-empty#inexistant
+[ERROR] Could not synchronize ${BASEDIR}/OPAM/pinning/.opam-switch/sources/pin-empty from "git+file://${BASEDIR}/pin-empty#inexistant":
+        Git repository seems just initialized, try again after your first commit
+Could not retrieve some package sources, they will not be pinned nor installed:
+  - pin-empty:git+file://${BASEDIR}/pin-empty#inexistant
+
+Continue anyway? [y/n] y
+### opam pin

